### ISSUE TITLE
Add alias for email in Resource Owner

### ISSUE THREAD

**Issue by jamesgraham** [TheNetworg/oauth2-azure#35]
Is there any harm adding `getEmail` as an alias of the `upn` claim?

Almost every oauth client packages implements `getEmail` so it would be useful to just have that available.

**Comment by hajekj** [comment]
I don't see any reason why it should be a problem. However, I did not originally add it, because in some AAD deployments (in common ones, this is completely fine), your UPN may be different than your e-mail and the `id_token` doesn't supply any specific e-mail attribute. The best way to get user's actual e-mail address is to obtain it from [MS Graph's /me endpoint](https://developer.microsoft.com/en-us/graph/docs/api-reference/v1.0/api/user_get).

**Comment by jamesgraham** [comment]
I was trying to avoid that extra lookup but it might be safer. Is there any documentation on where a UPN might not be an email address?

**Comment by hajekj** [comment]
I don't think that there is a documentation, but for instance, you can have users login with a short alias name like jahaj@domain.com and then use jan.hajek@domain.com as an e-mail address. Basically this can be very easily set in Exchange Online. Like I said, it is not a common scenario and when adding a new default address, EXO will automatically keep the original address (UPN in this case) as a secondary one. I have seen this kind of stuff with big corporations and multinational companies which use Office 365, but I believe that most of them allowed users to receive e-mail on the alias (UPN) as well.
